salt/modules/vault.py

```diff
@@ -13,352 +14,917 @@
+
+    .. note::
+
+        List members which do not have simple string representations,
+        such as dictionaries or objects, do not work and will
+        throw an exception. Strings and numbers are examples of
+        types which work well.
+
+cache_time
+    .. versionadded:: 3007.0
+
+    Number of seconds compiled templated policies are cached on the master.
+    This is important when using pillar values in templates, since compiling
+    the pillar is an expensive operation.
+
+    .. note::
+
+        Only effective when issuing tokens to minions. Token policies
+        need to be compiled every time a token is requested, while AppRole-associated
+        policies are written to Vault configuration the first time authentication data
+        is requested (they can be refreshed on demand by running
+        ``salt-run vault.sync_approles``).
+
+        They will also be refreshed in case other issuance parameters are changed
+        (such as uses/ttl), either on the master or the minion
+        (if allow_minion_override_params is True).
 
-        peer_run:
-            .*:
-                - vault.generate_token
+refresh_pillar
+    .. versionadded:: 3007.0
+
+    Whether to refresh the minion pillar when compiling templated policies
+    that contain pillar variables.
+    Only effective when issuing tokens to minions (see note on cache_time above).
+
+    - ``null`` (default) only compiles the pillar when no cached pillar is found.
+    - ``false`` never compiles the pillar. This means templated policies that
+      contain pillar values are skipped if no cached pillar is found.
+    - ``true`` always compiles the pillar. This can cause additional strain
+      on the master since the compilation is costly.
+
+    .. note::
+
+        Hardcoded to True when issuing AppRoles.
+
+        Using cached pillar data only (refresh_pillar=False) might cause the policies
+        to be out of sync. If there is no cached pillar data available for the minion,
+        pillar templates will fail to render at all.
+
+        If you use pillar values for templating policies and do not disable
+        refreshing pillar data, make sure the relevant values are not sourced
+        from Vault (ext_pillar, sdb) or from a pillar sls file that uses the vault
+        execution/sdb module. Although this will often work when cached pillar data is
+        available, if the master needs to compile the pillar data during policy rendering,
+        all Vault modules will be broken to prevent an infinite loop.
+
+``server``
+~~~~~~~~~~
+.. versionchanged:: 3007.0
+
+    The values found in here were found in the ``vault`` root namespace previously.
+
+Configures Vault server details.
+
+url
+    URL of your Vault installation. Required.
+
+verify
+    Configures certificate verification behavior when issuing requests to the
+    Vault server. If unset, requests will use the system default CA bundle.
```

Comment:
This is actually incorrect if `requests` was installed by `pip`, not via the (patched) system packages. In that case, it relies on the certificates found in the `certifi` package and ignores the system default CA bundle.

I have a patch that allows to specify an explicit root CA, which I would like to include here or in a follow-up PR.